src/lib/AutoControlledComponent.js

```diff
@@ -30,27 +30,33 @@ const getDefaultPropName = (prop) => `default${prop[0].toUpperCase() + prop.slic
 
 /**
  * Return the auto controlled state value for a give prop. The initial value is chosen in this order:
- *  - default props
- *  - then, regular props
+ *  - regular props
+ *  - then, default props
+ *  - then, initial state
  *  - then, `checked` defaults to false
  *  - then, `value` defaults to '' or [] if props.multiple
  *  - else, undefined
  *
- *  @param {object} props A props object
  *  @param {string} propName A prop name
- *  @param {boolean} [includeDefaultProps=false] Whether or not to heed the default prop value
+ *  @param {object} [props] A props object
+ *  @param {object} [state] A state object
+ *  @param {boolean} [includeDefaults=false] Whether or not to heed the default props or initial state
  */
-export const getAutoControlledStateValue = (props, propName, includeDefaultProps = false) => {
-  const defaultPropName = getDefaultPropName(propName)
-  const prop = props[propName]
-  const defaultProp = props[defaultPropName]
-
-  const hasProp = prop !== undefined
-  const hasDefaultProp = defaultProp !== undefined
-
-  // defaultProps & props
-  if (includeDefaultProps && !hasProp && hasDefaultProp) return defaultProp
-  if (hasProp) return prop
+export const getAutoControlledStateValue = (propName, props, state, includeDefaults = false) => {
```

Comment:
Updated this function to handle initial state.  Since this is the SSOT when it comes to how we derive AC state I think it makes sense here.  It also prevents an extra loop of state in CWM.